docs/page-caching.md

```diff
@@ -1,42 +1,74 @@
+
+There are 2 levels of page caching:
+
+- CDN caching, served from [regional cache nodes](docs://cloud/cdn/)
+- PHP caching, served from PHP using [Batcache](https://github.com/humanmade/batcache)
+
+The CDN cache is the fastest cache, and has typical response times around 20ms. If a page cannot be served from the CDN cache, it may fall back to the Batcache cache. This is served from PHP on the application server, and has typical response times around 100-200ms.
+
+All pages are cached by default, with the following exclusions:
 
 - All `POST` requests are not cached
 - Any requests with cookies matching the patterns `wordpress_*`, `wp-*`, `wp_*`, `comment_*`, `hm_*` and  `altis_*` are not cached.
 - Any requests with the `Authentication` HTTP header will not be cached.
 - Any response with the `Cache-Control: no-cache` HTTP header will not be cached.
-- Any response with no content are not cached.
+- Any response with no content is not cached.
 - Any request to `xmlrpc.php`, `wp-cron.php` or `wp-app.php` are not cached.
 - Any request to `wp-includes/js*` are not cached.
 
-There are 2 levels of page caching:
+**Note:** Page caching is disabled on local environments, and due to non-production environments requiring authentication, will likely not be active during testing and QA. See the [configuration](#configuration) section below for testing instructions.
 
-- CloudFront CDN
-- PHP caching using Redis via [Batcache](https://github.com/humanmade/batcache)
 
 ## Cache Key Calculation
 
-Each cache key for pages follows the pattern:
+Each page is stored within cache storage as identified by a cache key. The cache key follows the pattern:
 
 ```
 {method}:{protocol}:{host}:{path}:{query_params}
 ```
 
-## Cache Times
+Query parameters can be [customized](#customizations) to ignore certain query parameters which do not effect the page content.
+
+**Note:** The cache key for the CDN cache is fixed, and query parameters can not be ignored within the CDN cache. The CDN cache key cannot be changed.
+
+
+## Time-to-live and maximum age
+
+Cacheable pages have default time-to-live (TTL) values within the cache based on the response code:
+
+- Success responses (`200`): 300 seconds (5 minutes)
+- Client errors (`400`, except `404`): 10 seconds
+- Not found errors (`404`): 300 seconds (5 minutes)
+- Server errors (`5XX`): 300 seconds (5 minutes)
+
+Time-to-live values should be balanced between achieving high cache hit rates and ensuring fresh content is served.
+
+This can be adjusted by sending a [`Cache-Control: max-age=x` HTTP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#response_directives) within your response, which will override any defaults.
+
+For example, to specify a lifetime of one hour for a specific page:
+
+```php
+header( 'Cache-Control: max-age=' . HOUR_IN_SECONDS );
```

Comment:
This header is set by batcache, and I think we should recommend changing the batcache config rather than sending a header

Reply:
Batcache [parses it from the header](https://github.com/humanmade/batcache/blob/3230d4424556b3a1c4209d7548a2f2d9a2927907/advanced-cache.php#L275)

Reply:
Yeah I realise that, but it kinda sucks!

Reply:
Eh, I think it works just fine. If you manually set `$batcache->max_age`, it'll get clobbered by that anyway. Also, this is better than using the weird undocumented `$batcache` global half way through a page render.

Reply:
Yeah, but like the comment says, you have to also check when you can send headers, so it's pretty rough. I think the best solution is `page-cache.ttl` in the config. Baring that, I'd still go `$batcache` global over having to send output just to tell the cache to do something.

Reply:
I was documenting this specifically for a use case which requires dynamic setting of the max age based on the content (i.e. cache any posts older than a month for a full day instead of 5 mins).

You actually _don't_ have to check when you can send headers since the output is all buffered anyway, but it's best practice in PHP. If that's a sticking point, I can just remove it.

Reply:
Ok, imo it's a worse api, but if we go with that I don't have an issue with the dev warning.